Reduce iterations to two

jonodrew opened the pull request:
I've done my best to make this flexible by passing the number of iterations as an environment variable. That means that, if we want to deploy this code for anyone else, we can create a new `Stack` object with a different number of iterations
![Screenshot 2023-11-08 at 17 28 32](https://github.com/mentor-matching-online/mentor-match/assets/3410350/ea3552fe-a49e-488d-acfa-b0798150b13b)


I would also point out that, if users only want two iterations instead of the usual three, they could simply...not use the last set.

jonodrew committed:
Allow iterations to be set by an environment variable

This will allow us to generate different stacks for different apps

jonodrew committed:
Add the number of iterations to the environment

johnpeart commented:
Bizarrely, this is still outputting 3 matches for me, even after a clean build of the app.
Must be some kind of error with my local machine. The code I have in the local repo is the same as the remote repo, it's just still churning out 3 matches!

Your build evidently is working correctly though, based on the screenshot.

Before we merge this, I think we should make this an extra screen and a selectable option. 
Thoughts?

jonodrew commented:
Let's make sure it's working for you first. You'll need to set an environment variable for this to work. The easiest way to do this is probably through the Dockerfile - I can add something to the readme. The reason I've got the screenshot is because I deployed the whole thing into AWS with the environment variable set to two - that's what 96f133038452ec2ff688c316ee01d4a70fdea7fd does

jonodrew committed:
Add the number of iterations to the Dockerfile

This allows users running the system locally with docker-compose to alter the number of matches they want

jonodrew commented:
@johnpeart try with the new commit on this branch, and change the `ITERATIONS` value in docker-compose.yml

johnpeart commented:
No dice.

I've tried: 

1. deleting the docker container and rebuilding using `docker-compose up` – same result, 3 matches
2. clearing browser cache and then repeating 1 – same result, 3 matches
3. deleting the repo and cloning it again, and repeating 1 and 2 – same result, 3 matches

😕

jonodrew commented:
![Screenshot 2023-12-19 at 11 32 15](https://github.com/mentor-matching-online/mentor-match/assets/3410350/737f6ff1-6d31-4658-b2e7-9d0bf2c75b8d)
Bizarre - definitely working for me. You're definitely pulling this branch?

johnpeart commented:
Definitely the right branch, so it's clearly something weird with my set up.

Running `docker system prune` to remove everything and then rebuilding the container doesn't fix the issue *but* oddly, running `docker exec mentor-match-worker-1 env` to pull the environment variables picks up `ITERATIONS=2`. 

So it must be the right files; just building the wrong outputs.
